### DISCUSSION TITLE
proxying via SWA CLI is slow

### DISCUSSION THREAD

**Discussion by bzbetty** [Azure/static-web-apps-cli#554] [Q&A] [answered]
Using SWA proxying to a react site using vite.

If I access the vite site directly the site loads almost instantly
![image](https://user-images.githubusercontent.com/533131/189256787-ea63c5ca-08b7-4024-865c-2b5e30c1e800.png)

but when proxying via SWA it takes ~10 seconds

![image](https://user-images.githubusercontent.com/533131/189256829-e1322f52-7ee1-4950-aaa1-3c15b20d5be1.png)

I originally thought it might be a http 1 vs http 2 thing, but both are served using http 1/1 without SSL.

Looking at individual requests there's a huge difference that I can't figure out

![image](https://user-images.githubusercontent.com/533131/189256966-70b4d29a-1b6a-4d6f-950a-8df172b842e9.png)

Is this a limitation of SWA CLI or am i maybe doing something wrong in the config?

**Comment by manekinekko** [comment]
Hi, do you have a repo that I can use to investigate the issue?

**Reply by bzbetty** [reply]
made a small reproduction in this repo - https://github.com/bzbetty/staticwebapp-bug-repro

npm i + npm start 

Then look at the response times on localhost:3000 vs localhost:4280

**Reply by bzbetty** [reply]
@manekinekko is there anything else I can do to help with this?

**Reply by manekinekko** [reply]
Thank you, @bzbetty! I am planning on digging into this asap. Will keep you posted.

**Comment by manekinekko** [comment, marked as answer]
Hi @bzbetty so after investigating the issue, I found out about the cause. I'll send a fix that's gonna boost the performance of the local dev experience by ~70%. Thank you again for reporting this performance issue 👍

**Vite dev server:**
![image](https://user-images.githubusercontent.com/1699357/190126496-82ca0a34-4a0e-4850-a501-4c8c49b8d019.png)

**SWA CLI**
![image](https://user-images.githubusercontent.com/1699357/190126535-8a030e11-b2fd-4090-8c50-fe0add3a65f3.png)

Here is a graph showing the perf boost.
![image](https://user-images.githubusercontent.com/1699357/190602702-a22134bc-4436-4bc4-ba3d-3718f11b8994.png)

Please note that given how the SWA CLI local emulator is architecture, it's challenging for us to beat the framework dev server. This is why the numbers of the SWA CLI load time are slightly higher than the framework dev server HTTP server.

FWIW, when configuring the SWA CLI to connect to your dev server, we are basically proxying all requests to that dev server, and this obviously adds some extra time to the overall load time.

```mermaid
sequenceDiagram
    participant Browser
    participant SWACLI
    participant ViteCLI
    Browser->>SWACLI: GET index.html
    SWACLI->>ViteCLI: GET index.html
    ViteCLI->>ViteCLI: Read html content from disk
    ViteCLI->>SWACLI: html content
    SWACLI->>Browser: html content
```

**Reply by bzbetty** [reply]
That's amazing news, well done!
